server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -93,6 +98,7 @@ public SegmentReplicationTargetService(
         this.recoverySettings = recoverySettings;
         this.onGoingReplications = new ReplicationCollection<>(logger, threadPool);
         this.sourceFactory = sourceFactory;
+        this.pTermLinkedToCopiedFiles = UNASSIGNED_PRIMARY_TERM;
```

Comment:
This is a per node class, I think this value needs to be recorded per every replication group.

Reply:
How about using a shardId to primaryTerm map?